subprojects/logging/src/main/java/org/gradle/api/logging/configuration/LoggingConfiguration.java

```diff
@@ -45,6 +45,17 @@
     void setConsoleOutput(ConsoleOutput consoleOutput);
 
     /**
+     * Returns whether dry-run mode is enabled or not.
+     * Defaults to false.
+     */
+    boolean isDryRun();
+
+    /**
+     * Enables or disables dry-run (log but do not perform task actions) mode.
+     */
+    void setDryRun(boolean dryRun);
```

Comment:
Yes, I think the better option is to only have a new method `setDryRun(boolean)` on `LoggingManagerInternal`. If you go for the `AbstractCommandLineConverter` model then we wouldn't have to expose the new method to `LoggingConfiguration` which is preferable from my perspective.